crates/turbopack/src/module_options/mod.rs

```diff
@@ -469,10 +470,17 @@ impl ModuleOptionsVc {
             } else {
                 package_import_map_from_context("loader-runner", path)
             };
-            for (ext, loaders) in webpack_loaders_options.extension_to_loaders.iter() {
+            for (glob, rule) in webpack_loaders_options.rules.await?.iter() {
                 rules.push(ModuleRule::new(
                     ModuleRuleCondition::All(vec![
-                        ModuleRuleCondition::ResourcePathEndsWith(ext.to_string()),
+                        if !glob.contains("/") {
+                            ModuleRuleCondition::ResourceBasePathGlob(GlobVc::new(glob).await?)
+                        } else {
+                            ModuleRuleCondition::ResourcePathGlob {
+                                base: execution_context.project_path().await?,
+                                glob: GlobVc::new(glob).await?,
+                            }
+                        },
```

Comment:
Nit: to avoid a negated negate.
```suggestion
                        if glob.contains("/") {
                            ModuleRuleCondition::ResourcePathGlob {
                                base: execution_context.project_path().await?,
                                glob: GlobVc::new(glob).await?,
                            }
                        } else {
                            ModuleRuleCondition::ResourceBasePathGlob(GlobVc::new(glob).await?)
                        },
```